### ISSUE TITLE
Detection of cgroup mount points under CentOS 6

### ISSUE THREAD

**Issue by n0guest** [firehol/netdata#563]
Hi.

With default config netdata will fail to detect correct mount points for cgroup controllers, but only for CentOS. Just because CentOS use /cgroup/ instead of /sys/fs/cgroup (as it in Debian and Ubuntu).

Here is cgroups parameters from netdata.conf under CentOS 6.x:

		[plugin:cgroups]
			# cgroups plugin resources = yes
			# check for new cgroups every = 10
			# enable cpuacct stat = on demand
			# enable cpuacct usage = on demand
			# enable memory = on demand
			# enable blkio = on demand
			# path to /sys/fs/cgroup/cpuacct = /sys/fs/cgroup/cpuacct
			# path to /sys/fs/cgroup/blkio = /sys/fs/cgroup/blkio
			# path to /sys/fs/cgroup/memory = /sys/fs/cgroup/memory
			# max cgroups to allow = 500
			# max cgroups depth to monitor = 0
			# enable new cgroups detected at run time = yes

But real mount points for cgroup controllers looks like here:

		# grep cgroup /proc/mounts
		cgroup /cgroup/cpuset cgroup rw,relatime,cpuset 0 0
		cgroup /cgroup/cpu cgroup rw,relatime,cpu 0 0
		cgroup /cgroup/cpuacct cgroup rw,relatime,cpuacct 0 0
		cgroup /cgroup/memory cgroup rw,relatime,memory 0 0
		cgroup /cgroup/devices cgroup rw,relatime,devices 0 0
		cgroup /cgroup/freezer cgroup rw,relatime,freezer 0 0
		cgroup /cgroup/net_cls cgroup rw,relatime,net_cls 0 0
		cgroup /cgroup/blkio cgroup rw,relatime,blkio 0 0

It's not so important problem and can be easily avoid by overriding "path to xxx" in config:

		[plugin:cgroups]
			path to /sys/fs/cgroup/cpuacct = /cgroup/cpuacct
			path to /sys/fs/cgroup/blkio = /cgroup/blkio
			path to /sys/fs/cgroup/memory = /cgroup/memory

But after quick digging i've found function read_cgroup_plugin_configuration() in sys_fs_cgroup.c. And i guess, this issue can be fixed somewhere in logic of mountinfo_find_by_filesystem_mount_source() from proc_self_mountinfo.c. But it's just a guesses, nothing more.

PS: If you need any additional information, please, let me know and i'll get it.

**Comment by ktsaou** [comment]
netdata is using `/proc/mounts` to find the cgroups are.
Could you please post your `/proc/mounts`? This is mine:

```sh
$ cat /proc/mounts
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
udev /dev devtmpfs rw,nosuid,relatime,size=10240k,nr_inodes=999569,mode=755 0 0
devpts /dev/pts devpts rw,relatime,gid=5,mode=620,ptmxmode=000 0 0
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
/dev/sdd3 / btrfs rw,relatime,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=5,subvol=/ 0 0
tmpfs /run tmpfs rw,nodev,relatime,size=807264k,mode=755 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,nosuid,nodev,noexec,relatime 0 0
configfs /sys/kernel/config configfs rw,nosuid,nodev,noexec,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,nosuid,nodev,noexec,relatime 0 0
cgroup_root /sys/fs/cgroup tmpfs rw,nosuid,nodev,noexec,relatime,size=10240k,mode=755 0 0
openrc /sys/fs/cgroup/openrc cgroup rw,nosuid,nodev,noexec,relatime,release_agent=/lib64/rc/sh/cgroup-release-agent.sh,name=openrc 0 0
cpuset /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cpu /sys/fs/cgroup/cpu cgroup rw,nosuid,nodev,noexec,relatime,cpu 0 0
cpuacct /sys/fs/cgroup/cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpuacct 0 0
blkio /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
memory /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
devices /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
freezer /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
net_cls /sys/fs/cgroup/net_cls cgroup rw,nosuid,nodev,noexec,relatime,net_cls 0 0
net_prio /sys/fs/cgroup/net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_prio 0 0
hugetlb /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
pids /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
mqueue /dev/mqueue mqueue rw,nosuid,nodev,noexec,relatime 0 0
shm /dev/shm tmpfs rw,relatime,size=16777216k,nr_inodes=2097152 0 0
/dev/sdd1 /boot ext2 rw,noatime,data=ordered 0 0
/dev/md127 /data ext4 rw,relatime,stripe=256 0 0
tmpfs /tmp tmpfs rw,relatime,size=25165824k,nr_inodes=2097152 0 0
/dev/md125 /mnt/mdroot ext4 rw,relatime 0 0
binfmt_misc /proc/sys/fs/binfmt_misc binfmt_misc rw,nosuid,nodev,noexec,relatime 0 0
rpc_pipefs /var/lib/nfs/rpc_pipefs rpc_pipefs rw,relatime 0 0
nfsd /proc/fs/nfsd nfsd rw,nosuid,nodev,noexec,relatime 0 0
shm /var/tmp tmpfs rw,relatime,size=16777216k,nr_inodes=2097152 0 0
/dev/sdd3 /var/lib/docker/btrfs btrfs rw,relatime,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=5,subvol=/var/lib/docker/btrfs 0 0
shm /var/lib/docker/containers/abfde8a81bd50326be32cadb757b9649afc2a02c344ea202d7349002f8f54863/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
/dev/sdd3 /run/docker/libcontainerd/abfde8a81bd50326be32cadb757b9649afc2a02c344ea202d7349002f8f54863/rootfs btrfs rw,relatime,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=3760,subvol=/var/lib/docker/btrfs/subvolumes/a44362e340d3a253fc78be50dc9336a79423c155812e115b8b1503e6fab9fe5f 0 0
nsfs /run/docker/netns/71e228f8bf90 nsfs rw 0 0
tracefs /sys/kernel/debug/tracing tracefs rw,nosuid,nodev,noexec,relatime 0 0
none /run/user/1000 tmpfs rw,relatime,mode=700,uid=1000 0 0
```

**Comment by ktsaou** [comment]
sorry! It uses `/proc/self/mountinfo`

This is mine:
```sh
$ cat /proc/self/mountinfo
16 20 0:4 / /proc rw,nosuid,nodev,noexec,relatime - proc proc rw
17 20 0:6 / /dev rw,nosuid,relatime - devtmpfs udev rw,size=10240k,nr_inodes=999569,mode=755
18 17 0:13 / /dev/pts rw,relatime - devpts devpts rw,gid=5,mode=620,ptmxmode=000
19 20 0:16 / /sys rw,nosuid,nodev,noexec,relatime - sysfs sysfs rw
20 0 0:17 / / rw,relatime - btrfs /dev/sdd3 rw,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=5,subvol=/
21 20 0:19 / /run rw,nodev,relatime - tmpfs tmpfs rw,size=807264k,mode=755
22 19 0:20 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime - securityfs securityfs rw
23 19 0:7 / /sys/kernel/debug rw,nosuid,nodev,noexec,relatime - debugfs debugfs rw
24 19 0:21 / /sys/kernel/config rw,nosuid,nodev,noexec,relatime - configfs configfs rw
25 19 0:22 / /sys/fs/fuse/connections rw,nosuid,nodev,noexec,relatime - fusectl fusectl rw
26 19 0:23 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime - tmpfs cgroup_root rw,size=10240k,mode=755
27 26 0:24 / /sys/fs/cgroup/openrc rw,nosuid,nodev,noexec,relatime - cgroup openrc rw,release_agent=/lib64/rc/sh/cgroup-release-agent.sh,name=openrc
28 26 0:25 / /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime - cgroup cpuset rw,cpuset
29 26 0:26 / /sys/fs/cgroup/cpu rw,nosuid,nodev,noexec,relatime - cgroup cpu rw,cpu
30 26 0:27 / /sys/fs/cgroup/cpuacct rw,nosuid,nodev,noexec,relatime - cgroup cpuacct rw,cpuacct
31 26 0:28 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime - cgroup blkio rw,blkio
32 26 0:29 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime - cgroup memory rw,memory
33 26 0:30 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime - cgroup devices rw,devices
34 26 0:31 / /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime - cgroup freezer rw,freezer
35 26 0:32 / /sys/fs/cgroup/net_cls rw,nosuid,nodev,noexec,relatime - cgroup net_cls rw,net_cls
36 26 0:33 / /sys/fs/cgroup/net_prio rw,nosuid,nodev,noexec,relatime - cgroup net_prio rw,net_prio
37 26 0:34 / /sys/fs/cgroup/hugetlb rw,nosuid,nodev,noexec,relatime - cgroup hugetlb rw,hugetlb
38 26 0:35 / /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime - cgroup pids rw,pids
39 17 0:15 / /dev/mqueue rw,nosuid,nodev,noexec,relatime - mqueue mqueue rw
40 17 0:36 / /dev/shm rw,relatime - tmpfs shm rw,size=16777216k,nr_inodes=2097152
42 20 8:49 / /boot rw,noatime - ext2 /dev/sdd1 rw,data=ordered
43 20 9:127 / /data rw,relatime - ext4 /dev/md127 rw,stripe=256
44 20 0:38 / /tmp rw,relatime - tmpfs tmpfs rw,size=25165824k,nr_inodes=2097152
46 20 9:125 / /mnt/mdroot rw,relatime - ext4 /dev/md125 rw
47 16 0:43 / /proc/sys/fs/binfmt_misc rw,nosuid,nodev,noexec,relatime - binfmt_misc binfmt_misc rw
48 20 0:44 / /var/lib/nfs/rpc_pipefs rw,relatime - rpc_pipefs rpc_pipefs rw
49 16 0:45 / /proc/fs/nfsd rw,nosuid,nodev,noexec,relatime - nfsd nfsd rw
50 20 0:36 /var/tmp /var/tmp rw,relatime - tmpfs shm rw,size=16777216k,nr_inodes=2097152
51 20 0:17 /var/lib/docker/btrfs /var/lib/docker/btrfs rw,relatime - btrfs /dev/sdd3 rw,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=5,subvol=/var/lib/docker/btrfs
52 20 0:49 / /var/lib/docker/containers/abfde8a81bd50326be32cadb757b9649afc2a02c344ea202d7349002f8f54863/shm rw,nosuid,nodev,noexec,relatime - tmpfs shm rw,size=65536k
53 21 0:17 /var/lib/docker/btrfs/subvolumes/a44362e340d3a253fc78be50dc9336a79423c155812e115b8b1503e6fab9fe5f /run/docker/libcontainerd/abfde8a81bd50326be32cadb757b9649afc2a02c344ea202d7349002f8f54863/rootfs rw,relatime - btrfs /dev/sdd3 rw,compress=lzo,ssd,discard,space_cache,autodefrag,subvolid=3760,subvol=/var/lib/docker/btrfs/subvolumes/a44362e340d3a253fc78be50dc9336a79423c155812e115b8b1503e6fab9fe5f
118 21 0:3 net:[4026532356] /run/docker/netns/71e228f8bf90 rw - nsfs nsfs rw
56 23 0:9 / /sys/kernel/debug/tracing rw,nosuid,nodev,noexec,relatime - tracefs tracefs rw
45 21 0:39 / /run/user/1000 rw,relatime - tmpfs none rw,mode=700,uid=1000
```

**Comment by ktsaou** [comment]
On my test centos 7.2 I see this:

```sh
# cat /proc/self/mountinfo
18 60 0:17 / /sys rw,nosuid,nodev,noexec,relatime shared:6 - sysfs sysfs rw
19 60 0:3 / /proc rw,nosuid,nodev,noexec,relatime shared:5 - proc proc rw
20 60 0:5 / /dev rw,nosuid shared:2 - devtmpfs devtmpfs rw,size=241340k,nr_inodes=60335,mode=755
21 18 0:16 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:7 - securityfs securityfs rw
22 20 0:18 / /dev/shm rw,nosuid,nodev shared:3 - tmpfs tmpfs rw
23 20 0:11 / /dev/pts rw,nosuid,noexec,relatime shared:4 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
24 60 0:19 / /run rw,nosuid,nodev shared:21 - tmpfs tmpfs rw,mode=755
25 18 0:20 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:8 - tmpfs tmpfs ro,mode=755
26 25 0:21 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:9 - cgroup cgroup rw,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd
27 18 0:22 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:19 - pstore pstore rw
28 25 0:23 / /sys/fs/cgroup/net_cls rw,nosuid,nodev,noexec,relatime shared:10 - cgroup cgroup rw,net_cls
29 25 0:24 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime shared:11 - cgroup cgroup rw,blkio
30 25 0:25 / /sys/fs/cgroup/cpu,cpuacct rw,nosuid,nodev,noexec,relatime shared:12 - cgroup cgroup rw,cpuacct,cpu
31 25 0:26 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime shared:13 - cgroup cgroup rw,memory
32 25 0:27 / /sys/fs/cgroup/hugetlb rw,nosuid,nodev,noexec,relatime shared:14 - cgroup cgroup rw,hugetlb
33 25 0:28 / /sys/fs/cgroup/perf_event rw,nosuid,nodev,noexec,relatime shared:15 - cgroup cgroup rw,perf_event
34 25 0:29 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime shared:16 - cgroup cgroup rw,devices
35 25 0:30 / /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime shared:17 - cgroup cgroup rw,cpuset
36 25 0:31 / /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime shared:18 - cgroup cgroup rw,freezer
57 18 0:32 / /sys/kernel/config rw,relatime shared:20 - configfs configfs rw
60 1 253:1 / / rw,relatime shared:1 - ext4 /dev/vda1 rw,data=ordered
16 20 0:14 / /dev/mqueue rw,relatime shared:22 - mqueue mqueue rw
37 18 0:7 / /sys/kernel/debug rw,relatime shared:23 - debugfs debugfs rw
38 20 0:15 / /dev/hugepages rw,relatime shared:24 - hugetlbfs hugetlbfs rw
39 19 0:34 / /proc/sys/fs/binfmt_misc rw,relatime shared:25 - autofs systemd-1 rw,fd=29,pgrp=1,timeout=300,minproto=5,maxproto=5,direct
40 60 0:35 / /tmp rw shared:26 - tmpfs tmpfs rw
71 39 0:36 / /proc/sys/fs/binfmt_misc rw,relatime shared:27 - binfmt_misc binfmt_misc rw
73 24 0:37 / /run/user/0 rw,nosuid,nodev,relatime shared:28 - tmpfs tmpfs rw,size=50084k,mode=700
```

I also added some log to see if netdata is able to parse this info properly, and it is.
Please post your `proc/self/mountinfo` to trace the issue.

**Comment by n0guest** [comment]
First thing, i need to apologize. I've mentioned CentOS 7 by mistake. This problem exists only under CentOS 6.x. I've changed issue title as soon as i confirmed it.

And here is mountinfo, that you've asked (but from CentOS 6.x):

		# cat /proc/self/mountinfo 
		18 59 0:17 / /sys rw,nosuid,nodev,noexec,relatime shared:6 - sysfs sysfs rw,seclabel
		19 59 0:3 / /proc rw,nosuid,nodev,noexec,relatime shared:5 - proc proc rw,hidepid=2
		20 59 0:5 / /dev rw,nosuid shared:2 - devtmpfs devtmpfs rw,seclabel,size=32952168k,nr_inodes=8238042,mode=755
		21 18 0:16 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:7 - securityfs securityfs rw
		22 20 0:18 / /dev/shm rw,nosuid,nodev shared:3 - tmpfs tmpfs rw,seclabel
		23 20 0:11 / /dev/pts rw,nosuid,noexec,relatime shared:4 - devpts devpts rw,seclabel,gid=5,mode=620,ptmxmode=000
		24 59 0:19 / /run rw,nosuid,nodev shared:22 - tmpfs tmpfs rw,seclabel,mode=755
		25 18 0:20 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:8 - tmpfs tmpfs ro,seclabel,mode=755
		26 25 0:21 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:9 - cgroup cgroup rw,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd
		27 18 0:22 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:19 - pstore pstore rw
		28 25 0:23 / /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime shared:10 - cgroup cgroup rw,cpuset,clone_children
		29 25 0:24 / /sys/fs/cgroup/cpu,cpuacct rw,nosuid,nodev,noexec,relatime shared:11 - cgroup cgroup rw,cpuacct,cpu
		30 25 0:25 / /sys/fs/cgroup/perf_event rw,nosuid,nodev,noexec,relatime shared:12 - cgroup cgroup rw,perf_event
		31 25 0:26 / /sys/fs/cgroup/net_cls rw,nosuid,nodev,noexec,relatime shared:13 - cgroup cgroup rw,net_cls
		32 25 0:27 / /sys/fs/cgroup/hugetlb rw,nosuid,nodev,noexec,relatime shared:14 - cgroup cgroup rw,hugetlb
		33 25 0:28 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime shared:15 - cgroup cgroup rw,memory
		34 25 0:29 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime shared:16 - cgroup cgroup rw,blkio
		35 25 0:30 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime shared:17 - cgroup cgroup rw,devices
		36 25 0:31 / /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime shared:18 - cgroup cgroup rw,freezer
		57 18 0:32 / /sys/kernel/config rw,relatime shared:20 - configfs configfs rw
		59 1 9:127 / / rw,relatime shared:1 - ext4 /dev/md1 rw,seclabel,data=ordered
		37 18 0:15 / /sys/fs/selinux rw,relatime shared:21 - selinuxfs selinuxfs rw
		38 19 0:33 / /proc/sys/fs/binfmt_misc rw,relatime shared:23 - autofs systemd-1 rw,fd=25,pgrp=1,timeout=300,minproto=5,maxproto=5,direct
		39 18 0:7 / /sys/kernel/debug rw,relatime shared:24 - debugfs debugfs rw
		40 20 0:14 / /dev/mqueue rw,relatime shared:25 - mqueue mqueue rw,seclabel
		41 20 0:34 / /dev/hugepages rw,relatime shared:26 - hugetlbfs hugetlbfs rw,seclabel
		158 24 0:35 / /run/user/0 rw,nosuid,nodev,relatime shared:111 - tmpfs tmpfs rw,seclabel,size=6592308k,mode=700
		657 59 9:127 /var/www/development/dev.somedomain.tld/_asset /home/someuserXX/_asset rw,relatime shared:1 - ext4 /dev/md1 rw,seclabel,data=ordered
		1121 59 9:127 /var/www/development/dev.somedomain.tld/s /home/someuserXX/s rw,relatime shared:1 - ext4 /dev/md1 rw,seclabel,data=ordered
		1127 59 9:127 /var/www/development/dev.somedomain.tld/actions/shopkins /home/someuserXX/actions/shop rw,relatime shared:1 - ext4 /dev/md1 rw,seclabel,data=ordered
		1137 59 9:127 /var/www/development/dev.somedomain.tld/_ignore/html /home/someuserXX/_ignore/html rw,relatime shared:1 - ext4 /dev/md1 rw,seclabel,data=ordered
		1171 59 0:102 / /tmp rw shared:739 - tmpfs tmpfs rw,seclabel
		1027 24 0:100 / /run/user/995 rw,nosuid,nodev,relatime shared:654 - tmpfs tmpfs rw,seclabel,size=6592308k,mode=700,uid=995,gid=993

**Comment by ktsaou** [comment]
But this printout says: `/sys/fs/cgroup`.
?

**Comment by n0guest** [comment]
Ooh, sorry. It was a very long day. I've accidentally posted mountinfo from CentOS 7.x. Here is from server with CentOS 6.x:

		# uname -a
		Linux domain.tld 2.6.32-573.8.1.el6.x86_64 #1 SMP Tue Nov 10 18:01:38 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

		# cat /etc/redhat-release
		CentOS release 6.6 (Final)

		# cat /proc/self/mountinfo
		16 21 0:3 / /proc rw,relatime - proc proc rw,hidepid=2
		17 21 0:0 / /sys rw,relatime - sysfs sysfs rw
		18 21 0:5 / /dev rw,relatime - devtmpfs devtmpfs rw,size=8220400k,nr_inodes=2055100,mode=755
		19 18 0:11 / /dev/pts rw,relatime - devpts devpts rw,gid=5,mode=620,ptmxmode=000
		20 18 0:16 / /dev/shm rw,relatime - tmpfs tmpfs rw
		21 1 252:1 / / rw,relatime - ext4 /dev/vda1 rw,barrier=1,data=ordered,usrquota,grpquota
		22 16 0:15 / /proc/bus/usb rw,relatime - usbfs /proc/bus/usb rw
		23 16 0:17 / /proc/sys/fs/binfmt_misc rw,relatime - binfmt_misc none rw
		24 21 0:18 / /cgroup/cpuset rw,relatime - cgroup cgroup rw,cpuset
		25 21 0:19 / /cgroup/cpu rw,relatime - cgroup cgroup rw,cpu
		26 21 0:20 / /cgroup/cpuacct rw,relatime - cgroup cgroup rw,cpuacct
		27 21 0:21 / /cgroup/memory rw,relatime - cgroup cgroup rw,memory
		28 21 0:22 / /cgroup/devices rw,relatime - cgroup cgroup rw,devices
		29 21 0:23 / /cgroup/freezer rw,relatime - cgroup cgroup rw,freezer
		30 21 0:24 / /cgroup/net_cls rw,relatime - cgroup cgroup rw,net_cls
		31 21 0:25 / /cgroup/blkio rw,relatime - cgroup cgroup rw,blkio
		32 17 0:7 / /sys/kernel/debug rw,relatime - debugfs none rw

		# lssubsys -m
		cpuset /cgroup/cpuset
		cpu /cgroup/cpu
		cpuacct /cgroup/cpuacct
		memory /cgroup/memory
		devices /cgroup/devices
		freezer /cgroup/freezer
		net_cls /cgroup/net_cls
		blkio /cgroup/blkio

**Comment by ktsaou** [comment]
hm... it should be parsed correctly.
Could you please update to the latest netdata, restart it and run `grep cgroup /var/log/netdata/error.log` ?

If it cannot parse it, it should state `Cannot find cgroup ...`. Do you have such logs?

**Comment by n0guest** [comment]
Here is current cgroups on this server:

		# lscgroup
		cpuset:/
		cpu:/
		cpu:/php
		cpu:/mysql
		cpu:/apache
		cpu:/apache/user1
		cpu:/apache/user2
		cpu:/apache/user3
		cpu:/apache/user4
		cpuacct:/
		cpuacct:/php
		cpuacct:/mysql
		cpuacct:/apache
		cpuacct:/apache/user1
		cpuacct:/apache/user2
		cpuacct:/apache/user3
		cpuacct:/apache/user4
		memory:/
		memory:/php
		memory:/mysql
		memory:/apache
		memory:/apache/user1
		memory:/apache/user2
		memory:/apache/user3
		memory:/apache/user4
		devices:/
		freezer:/
		net_cls:/
		blkio:/
		blkio:/php
		blkio:/mysql
		blkio:/apache
		blkio:/apache/user1
		blkio:/apache/user2
		blkio:/apache/user3
		blkio:/apache/user4

Parameters for cgroup in resulting netdata.conf:

		[plugin:cgroups]
			# cgroups plugin resources = yes
			# check for new cgroups every = 10
			# enable cpuacct stat = on demand
			# enable cpuacct usage = on demand
			# enable memory = on demand
			# enable blkio = on demand
			# path to /sys/fs/cgroup/cpuacct = /sys/fs/cgroup/cpuacct
			# path to /sys/fs/cgroup/blkio = /sys/fs/cgroup/blkio
			# path to /sys/fs/cgroup/memory = /sys/fs/cgroup/memory
			# max cgroups to allow = 500
			# max cgroups depth to monitor = 0
			# enable new cgroups detected at run time = yes

I've truncated error log and restarted netdata. And part of error log:

		# grep -A10 -B10 -i cgroup error.log
		16-06-16 23:59:14: INFO: netdata: NetData exiting. Bye bye...
		16-06-16 23:59:14: INFO: netdata: Successfully became user 'root'.
		16-06-16 23:59:14: INFO: netdata: NetData started on pid 11262
		16-06-16 23:59:14: ERROR: netdata: Cannot create directory '/var/lib/netdata/registry'. Registry disabled. (errno 2, No such file or directory)
		16-06-16 23:59:14: INFO: netdata: Starting thread tc.
		16-06-16 23:59:14: INFO: netdata: Starting thread idlejitter.
		16-06-16 23:59:14: INFO: netdata: TC thread created with task id 11264
		16-06-16 23:59:14: INFO: netdata: Starting thread proc.
		16-06-16 23:59:14: INFO: netdata: CPU Idle Jitter thread created with task id 11265
		16-06-16 23:59:14: INFO: netdata: executing command: 'exec /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1' on pid 11266.
		16-06-16 23:59:14: INFO: netdata: Starting thread cgroups.
		16-06-16 23:59:14: INFO: netdata: PROC Plugin thread created with task id 11267
		16-06-16 23:59:14: INFO: netdata: 16-06-16 23:59:14: INFO: netdata: CGROUP Plugin thread created with task id 11268
		Starting thread plugins.d.
		16-06-16 23:59:14: INFO: netdata: PLUGINS.D thread created with task id 11269
		16-06-16 23:59:14: INFO: netdata: Not starting thread check.
		16-06-16 23:59:14: INFO: netdata: Starting thread web.
		16-06-16 23:59:14: INFO: netdata: WEB SERVER thread created with task id 11270
		16-06-16 23:59:14: INFO: netdata: executing command: 'exec /usr/libexec/netdata/plugins.d/apps.plugin 1 ' on pid 11272.
		16-06-16 23:59:14: INFO: netdata: PLUGINSD: '/usr/libexec/netdata/plugins.d/apps.plugin' running on pid 11272
		16-06-16 23:59:14: INFO: netdata: PLUGINSD: '/usr/libexec/netdata/plugins.d/node.d.plugin' running on pid 11274
		16-06-16 23:59:14: INFO: netdata: executing command: 'exec /usr/libexec/netdata/plugins.d/node.d.plugin 1 ' on pid 11274.
		16-06-16 23:59:14: INFO: netdata: PLUGINSD: '/usr/libexec/netdata/plugins.d/charts.d.plugin' running on pid 11277

> Could you please update to the latest netdata

Is it mandatory? This issue from server of our client and i would prefer don't mess with their production server. If you think this is only way or fastest way - i'll setup same environment in LXC container and do as you said.

PS: BTW, this is build from revision tagged as v1.2.0 (i.e. release of 1.2.0)

**Comment by ktsaou** [comment]
oh... I see. I think I fixed it after the 1.2.0 release.... let me check...

Yes, I fixed that on Apr 30th: https://github.com/firehol/netdata/commit/91a7221964e32ec4f727e7be5ee892b451affe28

So, newer netdata find out already.

**Comment by ktsaou** [comment]
Keep in mind you run your netdata as root. Generally you should not do that. Why did you need it?

**Comment by n0guest** [comment]
> Yes, I fixed that on Apr 30th: 91a7221

Thanks for inforomation. We'll' wait for next stable release.

> Keep in mind you run your netdata as root. Generally you should not do that. Why did you need it?

Because proc mounted with option hidepid=2. This mode is hiding full process list from all users except root. And when netdata started under unprivileged user, it can't see full process list. And this condition breaks "Application" statistics (it see only itself).

**Comment by ktsaou** [comment]
ok. Just protect your netdata then.

I am closing this issue. If you need something more, let me know.
